cvmfs/sync_union.h

```diff
@@ -34,6 +34,8 @@
 #include <string>
 
 #include "sync_item.h"
+#include "sync_mediator.h"
```

Comment:
If possible, a forward declaration of `AbstractSyncMediator` instead of the include would be preferable